src/broker/forwarding_table.rs

```diff
@@ -58,17 +58,34 @@ impl ForwardingTable {
                 0
             };
 
-            // walk the list of clients that are interested in the topic
+            if rx_count == 0 {
+                continue;
+            }
+
+            // fetch the topic
+            let topic = match String::from_utf8(buf[0].clone()) {
+                Ok(t) => t,
+                Err(_) => continue,
+            };
+
+            // retrieve the list of subscribers for this topic
+            let subscribers = match self.table.get(&topic) {
+                Some(subs) => subs,
+                None => continue,
+            };
+
+            // walk the list of clients that are subscribed to the topic
             // and forward the message to them
-            for d in self.publishers.iter() {
+            for d in subscribers.iter() {
+                // skip if the publisher is trying to send to itself
```

Comment:
useful comment

Reply:
Thinking about it more, when we move to the final "router" we may want to omit this check - should be no reason a client cannot send to itself